src/companion.jl

```diff
@@ -15,86 +16,78 @@ Also, directly from a polynomial:
 ```julia
 julia> using Polynomials
 
-julia> P=Polynomial([2.0,3,4,5])
+julia> P = Polynomial(2:5)
 Polynomial(2 + 3x + 4x^2 + 5x^3)
 
-julia> C=Companion(P)
+julia> C = Companion(P)
 3×3 Companion{Float64}:
  0.0  0.0  -0.4
  1.0  0.0  -0.6
  0.0  1.0  -0.8
 ```
 """
-struct Companion{T} <: AbstractArray{T, 2}
+struct Companion{T} <: AbstractMatrix{T}
     c :: Vector{T}
 end
 
-# Generate companion matrix from a polynomial
+Companion(v::AbstractVector) = Companion(collect(v))
 
-function Companion(P::Polynomial{T}) where T
-   n = length(P)
-   c = Array{T}(undef, n-1)
-   d=P.coeffs[n]
-   for i=1:n-1
-       c[i]=P.coeffs[i]/d
-   end
-   Companion(c)
+# Construct companion matrix from a polynomial
+
+function Companion(P::Polynomial)
+    n = length(P)
+    c = P.coeffs[1:(n-1)] / P.coeffs[n]
+    return Companion(c)
 end
 
-#Basic property computations
-size(C::Companion, r::Int) = (r==1 || r==2) ? length(C.c) :
-    throw(ArgumentError("Companion is of rank 2"))
```

Comment:
Can you prove this is redundant? Have you considered constant propagation and invalidations?

Reply:
I have added a test of the form `size(C,3) == 1` to verify that the defaults in Base for an AbstractArray do what is needed here.  I already had a test of the form `size(C,1) == n`.

The "proof" is that the [manual](https://docs.julialang.org/en/v1/manual/interfaces/#man-interface-array) for the AA interface specifies only that a single `size` function that returns a tuple is needed.
I have no idea about "constant propagation and invalidations."  (These are static structs.)